tools/bin/compile-survey.js

```diff
@@ -0,0 +1,290 @@
+#!/usr/bin/env node
+"use strict";
+
+const fs = require("fs");
+const path = require("path");
+
+const async = require("async");
+require("epipebomb")();
+const uniqBy = require("lodash.uniqby");
+const yaml = require("js-yaml");
+
+const argv = process.argv.slice(2);
+
+if (argv.length === 0) {
+  console.error("Usage: compile-survey /path/to/survey.yaml");
+  process.exit(1);
+}
+
+const surveyConfig = path.resolve(argv.shift());
+const BASE_PATH = path.dirname(surveyConfig);
+const PRESET_PATH = path.resolve(BASE_PATH, "..", "presets", "presets");
+const FIELD_PATH = path.resolve(BASE_PATH, "..", "presets", "fields");
+const OPTION_PATH = path.resolve(BASE_PATH, "..", "presets", "options");
+
+let surveyDefinition;
+
+try {
+  surveyDefinition = yaml.safeLoad(fs.readFileSync(surveyConfig));
+} catch (err) {
+  throw err;
+}
+
+const getParser = type => {
```

Comment:
Yup. Possibly even in the mobile app, depending how presets are distributed (to facilitate focused editing w/o needing much of a server component).